_posts/blog/2021-04-21-jshell-accumulo-feature.md

```diff
@@ -0,0 +1,141 @@
+Inside `jshell-init.jsh` contains [Accumulo Java APIs][public APIs] formatted as import statements 
+and [AccumuloClient][client] build implementation. On startup the script automatically loads in the 
+APIs and attempts to construct a client. Should additional APIs and/or code implementations be 
+needed, simply append them to `jshell-init.jsh`. Alternatively you can create a separate JShell 
+script and specify the custom script's file path on startup.
+
+The build implementation finds and uses `accumulo-client.properties` in Accumulo's 
+classpath to auto-generate an [AccumuloClient][client] called **client**. 
+If `accumulo-client.properties` is found, a similar result will be produced below:
+
+``` 
+Preparing JShell for Apache Accumulo 
+
+Building Accumulo client using 'jar:file:/home/accumulo/lib/accumulo-client.jar!/accumulo-client.properties'
+
+Use 'client' to interact with Accumulo
+
+|  Welcome to JShell -- Version 11.0.10
+|  For an introduction type: /help intro
+
+jshell> 
+```
+If `accumulo-client.properties` is not found, an [AccumuloClient][client] will not 
+auto-generate and will produce the following result below:
+
+```
+Preparing JShell for Apache Accumulo 
+
+'accumulo-client.properties' was not found on the classpath
+
+|  Welcome to JShell -- Version 11.0.10
+|  For an introduction type: /help intro
+
+jshell> 
+```
+
+## JShell Accumulo Example
+1) Booting up JShell using default script
+
+```
+Preparing JShell for Apache Accumulo 
+
+Building Accumulo client using 'file:/home/accumulo/conf/accumulo-client.properties'
+
+Use 'client' to interact with Accumulo
+
+|  Welcome to JShell -- Version 11.0.10
+|  For an introduction type: /help intro
+
+jshell> 
+```
+
+2) Providing JShell with an Accumulo task
+
+```Java
+  // Create a table called "GothamPD".
+  client.tableOperations().create("GothamPD");
+
+  // Create a Mutation object to hold all changes to a row in a table. 
+  // Each row has a unique row ID.
+  Mutation mutation = new Mutation("id0001");
+
+  // Create key/value pairs for Batman. Put them in the "hero" family.
+  mutation.put("hero", "alias", "Batman");
+  mutation.put("hero", "name", "Bruce Wayne");
+  mutation.put("hero", "wearsCape?", "true");
+
+  // Create a BatchWriter to the GothamPD table and add your mutation to it. 
+  // Try w/ resources will close for us.
+  try (BatchWriter writer = client.createBatchWriter("GothamPD")) {
+      writer.addMutation(mutation);
+  }
+
+  // Read and print all rows of the "GothamPD" table. 
+  // Try w/ resources will close for us.
+  try (org.apache.accumulo.core.client.Scanner scan =
+    client.createScanner("GothamPD", Authorizations.EMPTY)) {
+    System.out.println("Gotham Police Department Persons of Interest:");
+    
+    // A Scanner is an extension of java.lang.Iterable so behaves just like one.
+    for (Map.Entry<Key,Value> entry : scan) {
+      System.out.printf("Key : %-50s  Value : %s\n", entry.getKey(), entry.getValue());
+    }
```

Comment:
As of 2.1, we can use scanner.forEach, which makes this slightly nicer example code:

```suggestion
    scan.forEach((k, v) -> System.out.printf("Key : %-50s  Value : %s\n", k, v));
```